### DISCUSSION TITLE
Switch polling stops occurring after a period of time

### DISCUSSION THREAD

**Discussion by PeteRager** [zwave-js/zwave-js#6950] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?

After some time (~4 hours) application requests to poll switches are not being processed and the requests do not appear in the zwavejs log file.  Then later on they may start being processed.  This issue happens on zwavjs 9.14.1; but does not happen on zwavejs 9.10.3 or zwavejs 9.9.1.  I will be installing some newer version to see when the problem starts occurring.

The system is running on docker with Aeotec Gen5 FW 1.2 stick.

I have been running a combination on Home Assistant 2024.3.1 and Zwave JS 9.9.1 for 3 months with the polling from Home Assistant working correctly. I have automations that periodically poll switches on a staggered schedule every 30 minutes. on 6/19 I upgraded Zwavejs to 9.14.1 (as part of my quarterly updates); the polls worked correctly for several hours; then stopped overnight; and the resumed in the early morning hours; then failed again. The stick is active; updates are flowing in. So it seems to be limited to polling.  Im not 100% certain but it seemed like lights were still turning on / off from home assistant.  

I reviewed the Home Assistant Logs to verify the automations were running.  I then manually tried executing the service call

```
service: zwave_js.refresh_value
data:
  entity_id: switch.well
   refresh_all_values: false
```

The service call succeeded but zwavejs never processed the request and it never appears in the zwavejs logs. I then ran the current DEV branch of Home Assistant in the debugger (2024.6.3+) with it in this state and had the same behavior and I could verify that the messages were going out the websocket,  This failure mode seemed very odd; so I have spent time analyzing the best I can.

Since my goal was to be able to run HA 2024.6.3; I reviewed the docs to determine the minimum zwavejs version required which is 9.10.0; so I installed 9.10.3 and let it run overnite and there were no problems,  I plan to move up to the 9.11.x branch next to see if the problem occurs there.

Here is an example for Node 21; the polls run at :05 and 0:35

```
	Line    755: 2024-06-19T04:05:00.149Z DRIVER » [Node 021] [REQ] [SendData]   <- Running on 9.9.1
	Line   3629: 2024-06-19T04:35:00.103Z DRIVER » [Node 021] [REQ] [SendData]
	Line   6704: 2024-06-19T05:05:00.148Z DRIVER » [Node 021] [REQ] [SendData]
	Line   9982: 2024-06-19T05:35:00.101Z DRIVER » [Node 021] [REQ] [SendData]
	Line  13325: 2024-06-19T06:05:00.148Z DRIVER » [Node 021] [REQ] [SendData]
	Line  16502: 2024-06-19T06:35:00.100Z DRIVER » [Node 021] [REQ] [SendData]
	Line  20159: 2024-06-19T07:05:00.146Z DRIVER » [Node 021] [REQ] [SendData]
	Line  23472: 2024-06-19T07:35:00.100Z DRIVER » [Node 021] [REQ] [SendData]
	Line  27032: 2024-06-19T08:05:00.150Z DRIVER » [Node 021] [REQ] [SendData]
	Line  30301: 2024-06-19T08:35:00.104Z DRIVER » [Node 021] [REQ] [SendData]
	Line  33981: 2024-06-19T09:05:00.146Z DRIVER » [Node 021] [REQ] [SendData]
	Line  37076: 2024-06-19T09:35:00.100Z DRIVER » [Node 021] [REQ] [SendData]
	Line  40523: 2024-06-19T10:05:00.149Z DRIVER » [Node 021] [REQ] [SendData]
	Line  43495: 2024-06-19T10:35:00.102Z DRIVER » [Node 021] [REQ] [SendData]
	Line  47091: 2024-06-19T11:05:00.148Z DRIVER » [Node 021] [REQ] [SendData]
	Line  50210: 2024-06-19T11:35:00.101Z DRIVER » [Node 021] [REQ] [SendData]
	Line  53373: 2024-06-19T12:05:00.146Z DRIVER » [Node 021] [REQ] [SendData]
	Line  56359: 2024-06-19T12:35:00.103Z DRIVER » [Node 021] [REQ] [SendData]
	Line  59489: 2024-06-19T13:05:00.146Z DRIVER » [Node 021] [REQ] [SendData]
	Line  62818: 2024-06-19T13:35:00.102Z DRIVER » [Node 021] [REQ] [SendData]
	Line  66077: 2024-06-19T14:05:00.146Z DRIVER » [Node 021] [REQ] [SendData]
	Line  69049: 2024-06-19T14:35:00.103Z DRIVER » [Node 021] [REQ] [SendData]
	Line  72267: 2024-06-19T15:05:00.156Z DRIVER » [Node 021] [REQ] [SendData]
	Line  75418: 2024-06-19T15:35:00.101Z DRIVER » [Node 021] [REQ] [SendData]
	Line  78646: 2024-06-19T16:05:00.147Z DRIVER » [Node 021] [REQ] [SendData]
	Line  81878: 2024-06-19T16:35:00.102Z DRIVER » [Node 021] [REQ] [SendData]
	Line  85875: 2024-06-19T17:05:00.146Z DRIVER » [Node 021] [REQ] [SendData]
	Line  90413: 2024-06-19T17:35:00.101Z DRIVER » [Node 021] [REQ] [SendData]
	Line  94560: 2024-06-19T18:05:00.147Z DRIVER » [Node 021] [REQ] [SendData]
	Line  99314: 2024-06-19T18:35:00.101Z DRIVER » [Node 021] [REQ] [SendData]
	Line 103518: 2024-06-19T19:05:00.146Z DRIVER » [Node 021] [REQ] [SendData]
	Line 108415: 2024-06-19T19:35:00.104Z DRIVER » [Node 021] [REQ] [SendData]
	Line 112569: 2024-06-19T20:05:00.148Z DRIVER » [Node 021] [REQ] [SendData]
	Line 116709: 2024-06-19T20:35:00.102Z DRIVER » [Node 021] [REQ] [SendData]
	Line 118331: 2024-06-19T20:44:52.959Z DRIVER » [Node 021] [REQ] [SendData]  <- Upgraded to 9.14.1
	Line 118356: 2024-06-19T20:44:53.105Z DRIVER » [Node 021] [REQ] [SendData]
	Line 122487: 2024-06-19T21:05:00.414Z DRIVER » [Node 021] [REQ] [SendData]
	Line 126050: 2024-06-19T21:35:00.499Z DRIVER » [Node 021] [REQ] [SendData]
	Line 129466: 2024-06-19T22:05:00.414Z DRIVER » [Node 021] [REQ] [SendData]
	Line 133063: 2024-06-19T22:35:00.500Z DRIVER » [Node 021] [REQ] [SendData]
	Line 136645: 2024-06-19T23:05:00.414Z DRIVER » [Node 021] [REQ] [SendData]
	Line 139833: 2024-06-19T23:35:00.498Z DRIVER » [Node 021] [REQ] [SendData]
	Line 143656: 2024-06-20T00:05:00.414Z DRIVER » [Node 021] [REQ] [SendData]
	Line 143685: 2024-06-20T00:05:00.524Z DRIVER » [Node 021] [REQ] [SendData]
	Line 146826: 2024-06-20T00:35:00.498Z DRIVER » [Node 021] [REQ] [SendData]
	Line 150027: 2024-06-20T01:05:00.412Z DRIVER » [Node 021] [REQ] [SendData]
	Line 153521: 2024-06-20T01:35:00.498Z DRIVER » [Node 021] [REQ] [SendData]
	Line 157478: 2024-06-20T02:05:00.413Z DRIVER » [Node 021] [REQ] [SendData]
	Line 160743: 2024-06-20T02:35:00.499Z DRIVER » [Node 021] [REQ] [SendData]  <-  Gap starts
	Line 14591: 2024-06-20T07:05:00.413Z DRIVER » [Node 021] [REQ] [SendData]    <- Gap ends
	Line 17702: 2024-06-20T07:35:00.499Z DRIVER » [Node 021] [REQ] [SendData]
	Line 20812: 2024-06-20T08:05:00.412Z DRIVER » [Node 021] [REQ] [SendData]
	Line 24048: 2024-06-20T08:35:00.499Z DRIVER » [Node 021] [REQ] [SendData]
	Line 27312: 2024-06-20T09:05:00.414Z DRIVER » [Node 021] [REQ] [SendData]
	Line 30230: 2024-06-20T09:35:00.500Z DRIVER » [Node 021] [REQ] [SendData]
	Line 33683: 2024-06-20T10:05:00.414Z DRIVER » [Node 021] [REQ] [SendData]
	Line 37014: 2024-06-20T10:35:00.499Z DRIVER » [Node 021] [REQ] [SendData]     <- Problem starts
                            2024-06-20T16:40:29.958Z DRIVER   driver instance destroyed           <- shutdown container
```

What did you expect to happen instead?

The polls should work,

  2. Click on '...'
  3. Scroll down to '...'

Anything else we should know?

### Software versions

Driver (node-zwave-js): 12.1.1

Z-Wave JS UI: 9.14.1

Home Assistant Z-Wave Integration: 2024.3.1 and 2024.6.3.dev

Home Assistant Z-Wave JS Addon: n/a

ioBroker.zwave2 Adapter: n/a

If you are using something non-standard, tell us here: ...

### Device information

The issue effects all switches.  Here is a specific one that shows the problem

Manufacturer: Enbrighten
Model name:  46202 / ZW4009
Node ID: 21

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwavejs_2024-06-20.log](https://github.com/user-attachments/files/15926838/zwavejs_2024-06-20.log)

[zwavejs_2024-06-19.log](https://github.com/user-attachments/files/15926932/zwavejs_2024-06-19.log)

**Comment by PeteRager** [comment]
Just stared a test on will let it run for 24 hours and report back.  

zwave-js-ui: 9.13.3.61d406e
zwave-js: 12.9.1

**Reply by PeteRager** [reply]
Zwavejs 9.13.3 works fine. Let me know if I should try another version.

**Comment by AlCalzone** [comment]
It's possible that this is a duplicate of https://github.com/zwave-js/node-zwave-js/issues/6952
https://github.com/zwave-js/node-zwave-js/pull/6944 introduced a regression where sleeping nodes could block the send queue. A hotfix is coming. Please test again after you've installed the update.

**Comment by PeteRager** [comment]
Sounds likely.  I do launch staggered battery refreshes at 11 pm est, device wake-up every 4 hours - so that fits the first occurrence.

**Reply by AlCalzone** [reply]
FWIW,node-zwave-js also refreshes the battery if it hasn't received a reading in 7 days

**Comment by PeteRager** [comment]
Upgraded to 9.14.2 / 12.11.2 - will let run for 24 hours and report back.

**Reply by PeteRager** [reply, marked as answer]
The issue does not occur in zwavejs 19.4.2
